recipes/python/assets/create_oracle_asset.py

```diff
@@ -0,0 +1,247 @@
+parser.add_argument('-addRMANCatalog', required=False, default=0, choices=[0,1] help="If an RMAN catalog asset should be created")
+parser.add_argument('-rmanCatalogName', required=False, default="RMANCatalog", help="RMAN catalog name")
+parser.add_argument('-tnsName', required=False, default="rmancat", help="TNS name for RMAN catalog")
+parser.add_argument('-addCredentials', required=False,  choices=['wallet', 'oracle', 'os', 'osAndOracle'], help="Type of credential to be created for the generated oracle assets")
+parser.add_argument('-credentialName', required=False, help="Name of Oracle credentials to be assoiciated with the assets")
+
+args = parser.parse_args()
+
+nbserver = args.nbserver
+username = args.username
+password = args.password
+domainName = args.domainName
+domainType = args.domainType
+dbName = args.dbName
+addRMANCatalog = args.addRMANCatalog
+tnsName = args.tnsName
+rmanCatalogName = args.rmanCatalogName
+addCredentials = args.addCredentials
+credentialName = args.credentialName
+
+
+base_url = "https://" + nbserver + "/netbackup"
+asset_service_url = base_url + "/asset-service/queries/"
+credentials_url = base_url + "/config/credentials/"
+content_type = content_type = "application/vnd.netbackup+json;version=4.0"
+
+
+def create_oracle_db(databaseName, rmanCatalogId, credentialName):
+    manual_db_create_request = open_sample_payload("post_oracle_create_db.json")
+    if rmanCatalogId != '':
+        manual_db_create_request['data']['attributes']['parameters']['objectList'][0]['relationship'] = {"rmanCatalog":{"data": [rmanCatalogId]}}
+
+    if credentialName != '':
+        manual_db_create_request['data']['attributes']['parameters']['objectList'][0]['asset']['commonAssetAttributes']['credentials']= [{"credentialName": credentialName}]
+
+    manual_db_create_request['data']['attributes']['parameters']['objectList'][0]['asset']['databaseName']
+
+    oracle_create_db_response = requests.post(asset_service_url, headers=headers, json=manual_db_create_request, verify=False)
+
+    if oracle_create_db_response.status_code != 201:
+        print("\nAPI returned status code {}. Response: {}".format(oracle_create_db_response.status_code,
+                                                                 oracle_create_db_response.json()))
+        raise SystemExit("\nScript ended.\n")
+
+    print ("\nRequest to create Oracle DB has been posted.")
+
+    oracle_db_id_uri = verify_asset_creation(oracle_create_db_response.json()['data']['id'])
+
+    if oracle_db_id_uri != '':
+        print("\nOracle database created.")
+        oracle_db_id = oracle_db_id_uri.split('/')[-1]
+
+    manual_instance_create_request = open_sample_payload("post_oracle_create_instance.json")
+    manual_instance_create_request['data']['attributes']['parameters']['objectList'][0]['relationship']['database']['data'] = [oracle_db_id]

```

Comment:
If `oracle_db_id_uri` doesn't get assigned at 72, then this line will crash because `oracle_db_id` hasn't been assigned otherwise.

Reply:
I added a check to just end the script if we hit this case